betree/src/storage_pool/configuration.rs

```diff
@@ -300,6 +312,46 @@ impl LeafVdev {
                 mem,
                 format!("memory-{}", mem),
             )?)),
+            #[cfg(feature = "nvm")]
+            LeafVdev::PMemFile { .. } => {
+                let (path, len) = match self {
+                    LeafVdev::File(path) => unreachable!(),
+                    LeafVdev::FileWithOpts { .. } => unreachable!(),
+                    LeafVdev::Memory { .. } => unreachable!(),
+                    LeafVdev::PMemFile {path, len} => (path, len),
+                };
+
+                let mut is_pmem : i32 = 0;
+                let mut mapped_len : u64 = 0;
+                let mut file = match path.to_str() {
+                    Some(filepath_str) => match libpmem::pmem_file_open(format!("{}\0",filepath_str).as_str(), &mut mapped_len, &mut is_pmem) { //match libpmem::pmem_file_create(filepath_str, *len as u64, &mut mapped_len, &mut is_pmem) {
+                        Some(existing_handle) => existing_handle,
+                        None => match libpmem::pmem_file_create(format!("{}\0",filepath_str).as_str(), *len as u64, &mut mapped_len, &mut is_pmem) {//match libpmem::pmem_file_open(filepath_str, &mut mapped_len, &mut is_pmem) {
+                            Some(new_handle) => new_handle,
+                            None =>  panic!("{}", io::Error::new(io::ErrorKind::Other,
+                                                           format!("Failed to create or open handle for pmem file. Path: {}",
+                                                                   filepath_str)))
+                        }
```

Comment:
Avoid hard panics in this part of the code. Return the error via `Err(..)` instead.